conf/evolutions/default/6.sql

```diff
@@ -0,0 +1,112 @@
+
+# --- !Ups
+CREATE TABLE survey_category_option
+(
+  survey_category_option_id SERIAL NOT NULL,
+  survey_category_option_text TEXT NOT NULL,
+  PRIMARY KEY (survey_category_option_id)
+);
+
+CREATE TABLE survey_question
+(
+  survey_question_id SERIAL NOT NULL,
+  survey_question_text TEXT NOT NULL,
+  survey_input_type TEXT NOT NULL,
+  survey_category_option_id INT,
+  survey_display_rank INT,
+  deleted BOOLEAN NOT NULL DEFAULT FALSE,
+  survey_user_role_id INT NOT NULL DEFAULT 1,
+  required BOOLEAN NOT NULL DEFAULT FALSE,
+  PRIMARY KEY (survey_question_id),
+  FOREIGN KEY (survey_category_option_id) REFERENCES survey_category_option(survey_category_option_id)
+);
+
+CREATE TABLE survey_option(
+ survey_option_id INT NOT NULL,
+ survey_category_option_id INT NOT NULL,
+ survey_option_text TEXT NOT NULL,
+ survey_display_rank INT,
+ PRIMARY KEY (survey_option_id),
+ FOREIGN KEY (survey_category_option_id) REFERENCES survey_category_option(survey_category_option_id)
+);
+
+INSERT INTO survey_category_option VALUES (1, 'enjoyment');
+INSERT INTO survey_category_option VALUES (2, 'difficulty');
+INSERT INTO survey_category_option VALUES (3, 'self-efficacy');
+INSERT INTO survey_category_option VALUES (4, 'motivation');
+
+INSERT INTO survey_option VALUES (1, 1, 'Very boring', 1);
+INSERT INTO survey_option VALUES (2, 1, 'Boring', 2);
+INSERT INTO survey_option VALUES (3, 1, 'Neutral', 3);
+INSERT INTO survey_option VALUES (4, 1, 'Enjoyable', 4);
+INSERT INTO survey_option VALUES (5, 1, 'Very enjoyable', 5);
+INSERT INTO survey_option VALUES (6, 2, 'Very difficult', 1);
+INSERT INTO survey_option VALUES (7, 2, 'Difficult', 2);
+INSERT INTO survey_option VALUES (8, 2, 'Neutral', 3);
+INSERT INTO survey_option VALUES (9, 2, 'Easy', 4);
+INSERT INTO survey_option VALUES (10, 2, 'Very easy', 5);
+INSERT INTO survey_option VALUES (11, 3, 'Poor', 1);
+INSERT INTO survey_option VALUES (12, 3, 'Fair', 2);
+INSERT INTO survey_option VALUES (13, 3, 'Good', 3);
+INSERT INTO survey_option VALUES (14, 3, 'Very Good', 4);
+INSERT INTO survey_option VALUES (15, 3, 'Excellent', 5);
+INSERT INTO survey_option VALUES (16, 4, 'It''s fun.', 1);
```

Comment:
Haven't these been replaced? Its now a free form text field correct?

Reply:
These aren't directly mapped to the questions. These are mapped to the question category id to make them reusable.

Reply:
ok